packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts

```diff
@@ -0,0 +1,1105 @@
+import { gql } from "@apollo/client/core";
+import { addTypenameToDocument } from "@apollo/client/utilities";
+import { prepareEnvironment } from "@gmrchk/cli-testing-library";
+import { equal } from "@wry/equality";
+import { print, type DocumentNode } from "graphql";
+import { createHash } from "node:crypto";
+import { readFileSync } from "node:fs";
+import path from "node:path";
+import type { PersistedQueryManifestOperation } from "../index";
+
+test("prints help message with --help", async () => {
+  const { cleanup, runCommand } = await setup();
+
+  const { code, stdout } = await runCommand("--help");
+
+  expect(code).toBe(0);
+  expect(stdout).toMatchInlineSnapshot(`
+    [
+      "Usage: generate-persisted-query-manifest [options]",
+      "Generate a persisted query manifest file",
+      "Options:",
+      "-c, --config <path>  path to the config file",
+      "-v, --version        output the version number",
+      "-h, --help           display help for command",
+    ]
+  `);
+
+  await cleanup();
+});
+
+test("prints version number with --version", async () => {
+  const { cleanup, runCommand } = await setup();
+  const { version } = JSON.parse(
+    readFileSync(path.resolve(__dirname, "../../package.json"), "utf8"),
+  );
+
+  const { code, stdout } = await runCommand("--version");
+
+  expect(code).toBe(0);
+  expect(stdout).toEqual([version]);
+
+  await cleanup();
+});
+
+test("writes manifest file and prints location", async () => {
+  const { cleanup, exists, runCommand } = await setup();
+
+  const { code, stdout } = await runCommand();
+
+  expect(stdout).toEqual(["Manifest written to persisted-query-manifest.json"]);
+  expect(code).toBe(0);
+  expect(await exists("./persisted-query-manifest.json")).toBe(true);
+
+  await cleanup();
+});
+
+test("can extract operations from .graphql files", async () => {
+  const query = gql`
+    query GreetingQuery {
+      greeting
+    }
+  `;
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+
+  await writeFile("./src/query.graphql", print(query));
```

Comment:
Might be interesting to explicitly test parsing from an operation that isn't already formatted in print-output format (ie, we should see that it gets normalized)

Reply:
Good call! Will add a test case for that.

Reply:
Added in [af08635](https://github.com/apollographql/apollo-utils/pull/343/commits/af08635965329d03ba970bb5ef00184a34fce889)